src/jobs/response-interception.ts

```diff
@@ -0,0 +1,43 @@
+import { AxiosError, AxiosRequestConfig } from 'axios';
+
+import { httpService, IdentityService } from '@shared/api';
+import { createJob, TokenStorage } from '@shared/lib';
+
+type RequestConfig = AxiosRequestConfig<any> & {
+  retry?: boolean;
+};
+
+export default createJob(() => {
+  httpService.interceptors.response.use(
+    response => response,
+    async (error: AxiosError) => {
+      const config = error.config as RequestConfig;
+
+      if (error.response?.status === 401 && !config?.retry) {
+        config.retry = true;
+
+        const refreshToken = TokenStorage.getString('refreshToken');
+        const tokenType = TokenStorage.getString('tokenType');
+
+        if (!refreshToken || !tokenType) {
+          return Promise.reject(error);
```

Comment:
Shouldn't we reset tokens and redux/user here to force re-select isAuthenticated ?